paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc

```diff
@@ -453,6 +454,30 @@ class MatMulMKLDNNHandler
       matmul_attrs.set_output_scales(0, {scale_out});
     }
 
+    if (ctx.HasAttr("fuse_activation")) {
+      const auto fuse_activation = ctx.Attr<std::string>("fuse_activation");
+      const auto fuse_alpha =
+          ctx.HasAttr("fuse_alpha") ? ctx.Attr<float>("fuse_alpha") : 0.0f;
+      const auto fuse_beta =
+          ctx.HasAttr("fuse_beta") ? ctx.Attr<float>("fuse_beta") : 0.0f;
+      const auto activation_scale = ctx.HasAttr("activation_scale")
+                                        ? ctx.Attr<float>("activation_scale")
+                                        : 1.0f;
+      if (fuse_activation == "hard_sigmoid") {
+        post_operations.append_eltwise(activation_scale,
+                                       dnnl::algorithm::eltwise_linear,
+                                       fuse_alpha,
+                                       fuse_beta);
+        post_operations.append_eltwise(
+            activation_scale, dnnl::algorithm::eltwise_clip, 0.0f, 1.0f);
+      } else if (fuse_activation != "") {
+        const auto activation_algorithm =
+            paddle::platform::AcquireActivationAlgorithm(fuse_activation);
+        post_operations.append_eltwise(
+            activation_scale, activation_algorithm, fuse_alpha, fuse_beta);
+      }
+    }
+
     matmul_attrs.set_post_ops(post_operations);
     return matmul_attrs;
   }
```

Comment:
What do you think about creating a common static function which would append eltwises into post_ops object? That behavior is common between all operators that are fused with activation, and all these functions look exactly the same. By using single function we will be able to add support for more activation with less redundancy

Reply:
That's great idea. I will extend previously added `AcquireActivationAlgorithm` to append eltwises into post_ops, instead of just returning `dnnl::algorithm`